### PR TITLE
New tkinter GUI

### LINKED ISSUE #13: Updating the existing Tkinter based GUI
The current Tkinter GUI looks dull and basic. It should include better color schemes, better positioning and sizing of elements. It would be a great update to the current GUI :)

### PR DESCRIPTION
**Pull request by akshgpt7**
closes #13
I've come up with a dark-themed GUI for the App and applied it to all screens. Hope you like it. :)

### CODE REVIEW

**Comment by akshgpt7** [comment]
#14

**Comment by akshgpt7** [comment]
Issue #13

File: Attendance_app.py [hunk 1]
```diff
@@ -40,8 +40,7 @@ def __init__(self, *args, **kwargs):
 
         self.title_font = tkfont.Font(family='Helvetica', size=18, weight="bold", slant="italic")
         x, y = pyautogui.size()
-        geo = str(int(0.35*x))+'x'+str(int(0.85*y))
-        self.geometry(geo)
+        self.geometry("700x550")
```
Comment (Vibhu-Agarwal) [inline Attendance_app.py]: Converting dynamic window size to static would result in apparent different app window size for different screen sizes.
Consider reverting back to dynamic window size.

File: Attendance_app.py [hunk 2]
```diff
@@ -147,32 +146,50 @@ class StartPage(tk.Frame):
...
-        self.label.pack(side="top", fill="x", pady=10)
-        self.lb_class=tk.Label(self,text="CLASS-CODE : ",width=15)
-        self.lb_class.pack()
-        self.tv_class=tk.Entry(self,width=15)
+        self.bkg = '#222629'
+        self.text_color = '#65CCB8'
+        StartPage.config(self, bg=self.bkg)
+
+        self.label = tk.Label(self, width=25, text="Attendance Management", bg=self.bkg, fg="#F8E9A1", font=("Times",30))
+        self.label.place(x=90,y=40)
```
Comment (Vibhu-Agarwal) [inline Attendance_app.py]: Fix placement coordinates according to dynamic screen sizes (1)

File: Attendance_app.py [hunk 3]
```diff
@@ -147,32 +146,50 @@ class StartPage(tk.Frame):
...
+
+        self.lb_class = tk.Label(self, text="CLASS-CODE: ", bg=self.bkg, fg=self.text_color, font=("Courier",18))
+        self.lb_class.place(x=130, y=150)
```
Comment (Vibhu-Agarwal) [inline Attendance_app.py]: Fix placement coordinates according to dynamic screen sizes (2)

File: Attendance_app.py [hunk 4]
```diff
@@ -147,32 +146,50 @@ class StartPage(tk.Frame):
...
-        button1 = tk.Button(self, text="Go to Student Portal", width=30,
-                            command=lambda: self.doWork("StudentPanelPage",self.tv_class.get(),self.tv_username.get(),self.tv_pass.get()))
-        button2 = tk.Button(self, text="Go to Manager Portal", width=30,
-                            command=lambda: self.doWork("ManagerPanelPage",self.tv_class.get(),self.tv_username.get(),self.tv_pass.get()))
-        button1.pack()
-        button2.pack()
-        bt_exit = tk.Button(self, text="Exit",width=20,
-                           command=self.exit)
-        bt_exit.pack()
+        self.tv_class.place(x=300, y=155)
```
Comment (Vibhu-Agarwal) [inline Attendance_app.py]: Fix placement coordinates according to dynamic screen sizes (3)

File: Attendance_app.py [hunk 5]
```diff
@@ -147,32 +146,50 @@ class StartPage(tk.Frame):
...
+
+        self.lb_username = tk.Label(self, text="USERNAME : ", bg=self.bkg, fg=self.text_color, font=("Courier",18))
+        self.lb_username.place(x=130,y=200)
+        self.tv_username = tk.Entry(self,width=30)
+        self.tv_username.place(x=300,y=205)
```
Comment (Vibhu-Agarwal) [inline Attendance_app.py]: Fix placement coordinates according to dynamic screen sizes (4)

File: Attendance_app.py [hunk 6]
```diff
@@ -147,32 +146,50 @@ class StartPage(tk.Frame):
...
-                            command=lambda: self.doWork("ManagerPanelPage",self.tv_class.get(),self.tv_username.get(),self.tv_pass.get()))
-        button1.pack()
-        button2.pack()
-        bt_exit = tk.Button(self, text="Exit",width=20,
-                           command=self.exit)
-        bt_exit.pack()
+        self.tv_class.place(x=300, y=155)
+
+        self.lb_username = tk.Label(self, text="USERNAME : ", bg=self.bkg, fg=self.text_color, font=("Courier",18))
+        self.lb_username.place(x=130,y=200)
```
Comment (Vibhu-Agarwal) [inline Attendance_app.py]: Fix placement coordinates according to dynamic screen sizes (4)

File: Attendance_app.py [hunk 7]
```diff
@@ -147,32 +146,50 @@ class StartPage(tk.Frame):
...
+        self.tv_username = tk.Entry(self,width=30)
+        self.tv_username.place(x=300,y=205)
+
+        self.lb_pass = tk.Label(self, text="PASSWORD : ", bg=self.bkg, fg=self.text_color, font=("Courier",18))
+        self.lb_pass.place(x=130, y=250)
```
Comment (Vibhu-Agarwal) [inline Attendance_app.py]: Fix placement coordinates according to dynamic screen sizes (6)

File: Attendance_app.py [hunk 8]
```diff
@@ -147,32 +146,50 @@ class StartPage(tk.Frame):
...
+        self.tv_pass = tk.Entry(self, show="*", width=30)
+        self.tv_pass.place(x=300, y=255)
```
Comment (Vibhu-Agarwal) [inline Attendance_app.py]: Fix placement coordinates according to dynamic screen sizes (7)

File: Attendance_app.py [hunk 9]
```diff
@@ -147,32 +146,50 @@ class StartPage(tk.Frame):
...
+        self.showbtn = tk.Button(self, text="SHOW", bg="#ed3833", command=self.show)    #32ff6a
+        self.show = False
+        self.showbtn.place(x=555, y=253)
```
Comment (Vibhu-Agarwal) [inline Attendance_app.py]: Fix placement coordinates according to dynamic screen sizes (8)

File: Attendance_app.py [hunk 10]
```diff
@@ -147,32 +146,50 @@ class StartPage(tk.Frame):
...
+
+        self.button1 = tk.Button(self, bg="#45056e", fg=self.text_color, text="Go to Student Portal", width=16,height=3, font=("",15),
+                                 command=lambda: self.doWork("StudentPanelPage",self.tv_class.get(),self.tv_username.get(),self.tv_pass.get()))
+        self.button2 = tk.Button(self, bg="#5f1854", fg=self.text_color, text="Go to Manager Portal", width=16,height=3, font=("",15),
+                                 command=lambda: self.doWork("ManagerPanelPage",self.tv_class.get(),self.tv_username.get(),self.tv_pass.get()))
+        self.button1.place(x=100,y=350)
+        self.button2.place(x=360,y=350)
```
Comment (Vibhu-Agarwal) [inline Attendance_app.py]: Fix placement coordinates according to dynamic screen sizes (9)

File: Attendance_app.py [hunk 11]
```diff
@@ -147,32 +146,50 @@ class StartPage(tk.Frame):
...
+        self.bt_exit = tk.Button(self, bg="red", fg="yellow", text="Exit", width=10, command=self.exit)
+        self.bt_exit.place(x=300,y=500)
```
Comment (Vibhu-Agarwal) [inline Attendance_app.py]: Fix placement coordinates according to dynamic screen sizes (10)

File: Attendance_app.py [hunk 12]
```diff
@@ -147,32 +146,50 @@ class StartPage(tk.Frame):
...
+
+    def show(self):
+        if self.show==False:
+            self.tv_pass.config(show="")
+            self.show=True
+            self.showbtn.config(bg="#32ff6a")
+        elif self.show==True:
+            self.tv_pass.config(show="*")
+            self.show=False
+            self.showbtn.config(bg="#ed3833")
```
Comment (Vibhu-Agarwal) [inline Attendance_app.py]: Format code according to [PEP8](https://www.python.org/dev/peps/pep-0008/) and [Variable Annotations](https://www.python.org/dev/peps/pep-0008/#variable-annotations).

**Comment by akshgpt7** [comment]
@Vibhu-Agarwal , I've made the screen size and positioning dynamic. Kindly review.

**Comment by akshgpt7** [comment]
@Vibhu-Agarwal  
This tkinter gui is more dynamically adaptive with pep-8 violations fixed.

File: Attendance_app.py [hunk 13]
```diff
@@ -1,14 +1,14 @@
 import os
 import sys
+import pyautogui
```
Comment (Vibhu-Agarwal) [inline Attendance_app.py]: Move `pyautogui` [back] below standard imports.

File: Attendance_app.py [hunk 14]
```diff
@@ -1,14 +1,14 @@
 import os
 import sys
+import pyautogui
 import subprocess
 from datetime import datetime
 import tkinter as tk
-from tkinter import messagebox, END
```
Comment (Vibhu-Agarwal) [inline Attendance_app.py]: Use **`messagebox`** from **`tkinter`** in the import itself and use **`messagebox`** variable directly in the code (wherever used) as currently done in the master.
Some IDEs fail to recognize the function in the tkinter library, thus indicating a warning.

File: Attendance_app.py [hunk 15]
```diff
@@ -22,27 +22,46 @@
...
 current_class = 'Marauders'
-if current_class is not 'Marauders':
+if current_class != 'Marauders':
     current_class_obj = MainFile(current_class)
     FaceTrainObj = FaceTrain(current_class)
 else:
     current_class_obj = None
     FaceTrainObj = None
 
+"""Dynamic variables"""
```
Comment (Vibhu-Agarwal) [inline Attendance_app.py]: Replace string with a single-line comment

File: Attendance_app.py [hunk 16]
```diff
@@ -22,27 +22,46 @@
...
 x, y = pyautogui.size()
 
+percentage_width = x/(normal_width/100)
+percentage_height = y/(normal_height/100)
+scale_factor = ((percentage_width + percentage_height)/2)/100
+
+title_fontsize = int(30 * scale_factor)
+minimum_size = 24
+if title_fontsize < minimum_size:
+    title_fontsize = minimum_size
```
Comment (Vibhu-Agarwal) [inline Attendance_app.py]: Prefer using [max()](https://docs.python.org/3/library/functions.html#max)

File: Attendance_app.py [hunk 17]
```diff
@@ -22,27 +22,46 @@
...
+
+text_fontsize = int(18 * scale_factor)
+minimum_textsize = 13
+if text_fontsize < minimum_textsize:
+    text_fontsize = minimum_textsize
```
Comment (Vibhu-Agarwal) [inline Attendance_app.py]: Prefer using [max()](https://docs.python.org/3/library/functions.html#max)

File: Attendance_app.py [hunk 18]
```diff
@@ -148,45 +172,75 @@ def __init__(self, parent, controller):
...
-        self.label = tk.Label(self, width=25, text="Attendance Management", bg=self.bkg, fg="#F8E9A1", font=("Times",30))
-        self.label.place(x=x/21.34,y=y/27)
-
-        self.lb_class = tk.Label(self, text="CLASS-CODE: ", bg=self.bkg, fg=self.text_color, font=("Courier",18))
-        self.lb_class.place(x=x/14.77, y=y/7.2)
-        self.tv_class = tk.Entry(self, width=30)
+        entry_width = int(30 * scale_factor)
+        min_entrywidth = 25
+        if entry_width < min_entrywidth:
+            entry_width = min_entrywidth
```
Comment (Vibhu-Agarwal) [inline Attendance_app.py]: Prefer using [max()](https://docs.python.org/3/library/functions.html#max)

File: Attendance_app.py [hunk 19]
```diff
@@ -148,45 +172,75 @@ def __init__(self, parent, controller):
...
-        self.button2.place(x=x/5.34,y=y/3.09)
-        self.bt_exit = tk.Button(self, bg="red", fg="yellow", text="Exit", width=10, command=self.exit)
-        self.bt_exit.place(x=x/6.4,y=y/2.16)
+        self.showbtn.place(x=(340 * scale_factor) + (entry_width * 9),
+                           y=270*scale_factor)
+
+        bt_width = int(16 * scale_factor)
+        min_width = 12
+        if bt_width < min_width:
+            bt_width = min_width
```
Comment (Vibhu-Agarwal) [inline Attendance_app.py]: Prefer using [max()](https://docs.python.org/3/library/functions.html#max)

File: Attendance_app.py [hunk 20]
```diff
@@ -148,45 +172,75 @@ def __init__(self, parent, controller):
...
+        self.button1 = tk.Button(
+                     self, bg="#45056e", fg=self.text_color,
+                     text="Go to\nStudent Portal", width=bt_width, height=3,
+                     font=("", 15), command=lambda: self.doWork(
```
Comment (Vibhu-Agarwal) [inline Attendance_app.py]: Wrong hanging indentation (remove 9 spaces)

File: Attendance_app.py [hunk 21]
```diff
@@ -242,51 +298,72 @@ def __init__(self, parent, controller):
...
-        bt_exit = tk.Button(self, bg="#af0404", fg="yellow", text="Exit", width=10, command=self.exit)
-        bt_exit.place(x=x/3.80, y=y/2.16)
+        label = tk.Label(self, width=25, text="Manager Portal", bg=self.bkg,
+                         fg="#F8E9A1", font=("Times", title_fontsize))
+        label.place(x=130 * scale_factor, y=40 * scale_factor)
+
+        bt_ht = int(5 * scale_factor)
+        min_ht = 3
+        if bt_ht < min_ht:
+            bt_ht = min_ht
```
Comment (Vibhu-Agarwal) [inline Attendance_app.py]: Prefer using [max()](https://docs.python.org/3/library/functions.html#max)

File: Attendance_app.py [hunk 22]
```diff
@@ -242,51 +298,72 @@ def __init__(self, parent, controller):
...
+        bt_width = int(15 * scale_factor)
+        min_width = 12
+        if bt_width < min_width:
+            bt_width = min_width
```
Comment (Vibhu-Agarwal) [inline Attendance_app.py]: Prefer using [max()](https://docs.python.org/3/library/functions.html#max)

File: Attendance_app.py [hunk 23]
```diff
@@ -306,193 +383,249 @@ def __init__(self, parent, controller):
...
-        self.tv_class = tk.Entry(self, width=30, justify='center')
+        self.label = tk.Label(self, width=25, text="Create a new Batch",
+                              bg=self.bkg, fg="#F8E9A1",
+                              font=("Times", title_fontsize))
+        self.label.place(x=130 * scale_factor, y=40 * scale_factor)
+
+        entry_width = int(30 * scale_factor)
+        min_entrywidth = 23
+        if entry_width < min_entrywidth:
+            entry_width = min_entrywidth
```
Comment (Vibhu-Agarwal) [inline Attendance_app.py]: Prefer using [max()](https://docs.python.org/3/library/functions.html#max)

File: Attendance_app.py [hunk 24]
```diff
@@ -306,193 +383,249 @@ def __init__(self, parent, controller):
...
+        self.label.place(x=130 * scale_factor, y=40 * scale_factor)
+
+        self.lb_name = tk.Label(self, text="Name of Student: ",
+                                bg=self.bkg, fg=self.text_color,
+                                font=("Courier", text_fontsize))
+        self.lb_name.place(x=285 * scale_factor, y=175 * scale_factor)
+        entry_width = int(30 * scale_factor)
+        min_entrywidth = 23
+        if entry_width < min_entrywidth:
+            entry_width = min_entrywidth
```
Comment (Vibhu-Agarwal) [inline Attendance_app.py]: Prefer using [max()](https://docs.python.org/3/library/functions.html#max)

File: Attendance_app.py [hunk 25]
```diff
@@ -306,193 +383,249 @@ def __init__(self, parent, controller):
...
+            ws.cell(row=row_here, column=2,
+                    value=current_class + roll_no).fill = fill_roll
+            ws.cell(row=row_here, column=3,
+                    value=student_name).fill = fill_name
+            cellrange = 'A' + str(row_here) + ':AN' + str(row_here)
+            excel.set_border(ws, cellrange)
+            first_date_column = excel.get_column_letter(excel.date_column)
+            days_in_month = monthrange(today.year, today.month)[1]
+            last_date_column = excel.get_column_letter(
+                             excel.date_column + days_in_month - 1)
```
Comment (Vibhu-Agarwal) [inline Attendance_app.py]: 586 Wrong hanging indentation (remove 13 spaces)

File: Attendance_app.py (hunk 16 above)
Comment (akshgpt7) [reply Attendance_app.py]: I think max() function won't work here because there is no iterable to operate it on.

**Comment by akshgpt7** [comment]
@Vibhu-Agarwal All requested changes made. Please review.

**Comment by akshgpt7** [comment]
@Vibhu-Agarwal  All requested changes made. Please review.

**Comment by Vibhu-Agarwal** [comment]
Closes #13

**Comment by Vibhu-Agarwal** [comment]
@akshgpt7 Edit your PR description (first comment) with `resolves #13` as the first line.

**Comment by akshgpt7** [comment]
Sure, I've edited the description.

**Comment by Vibhu-Agarwal** [comment]
> Sure, I've edited the description.

You've edited your second comment and not the PR description.
Keywords like [these](https://help.github.com/en/github/managing-your-work-on-github/closing-issues-using-keywords) only work on PR description.

For reference, it's this comment of yours which needs to be edited
> I've come up with a dark-themed GUI for the App and applied it to all screens. Hope you like it. :)

**Pull request merged**